modules/openapi-generator/src/test/java/org/openapitools/codegen/python/PythonClientTest.java

```diff
@@ -439,6 +439,20 @@ public void testObjectWithValidations() {
         Assert.assertEquals((int) model.getMinProperties(), 1);
     }
 
+    @Test(description = "tests RegexObjects")
+    public void testRegexObjects() {
+        final OpenAPI openAPI = TestUtils.parseFlattenSpec("src/test/resources/3_0/issue_11521.yaml");
+        final DefaultCodegen codegen = new PythonClientCodegen();
+        codegen.setOpenAPI(openAPI);
+
+        String modelName = "DateTimeObject";
+        Schema modelSchema = ModelUtils.getSchema(openAPI, modelName);
+        final CodegenModel model = codegen.fromModel(modelName, modelSchema);
+        final CodegenProperty property1 = model.vars.get(0);
+        Assert.assertEquals(property1.baseName, "datetime");
+        Assert.assertEquals(property1.pattern, "/[\\d]{4}-[\\d]{2}-[\\d]{2}T[\\d]{1,2}:[\\d]{2}Z/");
```

Comment:
Please also check the value of pattern.vendorExtensions.x-regex
That is what the python client uses for its regex data per:
https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/python/model_templates/validations.mustache#L27

Reply:
This test fails (and it shouldn't) unless you remove the code in PythonLegacyClientCodegen.java. See #11521 for a description of the expected result and the actual result that comes out of the Python code generator for this regex

Reply:
A similar test (much more simple) that would also fail would be something like "a hex character", with a regex of: `"/[0-9a-fA-F]/"`

Reply:
Right; I agree that the test should be done. My ask here is that you also check the value of pattern.vendorExtensions.x-regex because that is what our template uses. Can you add a check of that value?
the pattern property that you are checking is not used in the generated code.

Reply:
Done!

Reply:
Thanks!